src/hgvs/utils/altseqbuilder.py

```diff
@@ -192,8 +194,17 @@ def _get_variant_region(self):
             and self._var_c.posedit.pos.end.datum == Datum.CDS_END
         ):
             result = self.WHOLE_GENE
+        elif self._var_c.posedit.edit.type == "ins" and self._var_c.posedit.pos.start.offset == -1 and self._var_c.posedit.pos.end.offset == 0:
+            # ins at intron-exon boundary
+            result = self.EXON
+        elif self._var_c.posedit.edit.type == "ins" and self._var_c.posedit.pos.start.offset == 0 and self._var_c.posedit.pos.end.offset == 1:
+            # ins at exon-intron boundary
+            result = self.EXON
+        elif self._var_c.posedit.edit.type == "dup" and self._var_c.posedit.pos.start.offset <= -1 and self._var_c.posedit.pos.end.offset >= -1:
```

Comment:
What cases of boundary-spanning dups are intended in this conditional?

I believe that the intended case is a dup that spans an intron-exon boundary (i.e., on the 5' end of an exon). However, I believe that misses a set of these cases and may match unintended cases. I've summarized these here:

![image](https://github.com/biocommons/hgvs/assets/109453/1363abd5-07c1-477a-8668-b4de91a91471)

Please comment on the following:

- `self._var_c.posedit.pos.end.offset >= -1`: Why not `== 0`, which is more specific? I believe that end is intended to be in the CDS. 
- Which of the cases do you intend to catch in the figure?

I'm happy to talk more about this interactively if useful.

Reply:
@reece Thanks for the comment.  The intention in this PR is to handle insertions that are directly at the boundary (not spanning the boundary).   So `self._var_c.posedit.pos.end.offset >= -1` should be updated to `self._var_c.posedit.pos.end.offset = -1` (because a duplication in the left intron will insert material directly at the boundary if the end position is -1). 

So in your image: 
1. Not intended to be handled (fixed with the change above)
5. Not intended to be handled.   And, no special logic is needed for the right end of the exon, because a duplication that inserts material at the boundary has no offsets, so is already considered exonic. 

2, 3, 4.  Good questions, we are thinking about these.